docs/content/en/docs/Upgrade/kubernetes.md

```diff
@@ -72,6 +72,70 @@ system-upgrade   apply-os-upgrade-on-kairos-with-1a1a24bcf897bd275730bdd8548-h7f
 
 Done! We should have all the basics to get our first cluster rolling, but there is much more we can do.
 
+## Verify images attestation during upgrades
+
+Container images can be signed during the build phase of a CI/CD pipeline using [Cosign](https://github.com/sigstore/cosign), Kairos signs every artifact as part of the release process.
+
+To ensure that the images used during upgrades match the expected signatures, [Kyverno](https://kyverno.io/) can be used to set up policies. This is done by checking if the signature is present in the OCI registry and if the image was signed using the specified key. The policy rule check fails if either of these conditions is not met.
+
+To learn more about this specific Kyverno feature, you can refer to the [documentation](https://kyverno.io/docs/writing-policies/verify-images/). This allows for the verification of image authenticity directly at the node level prior to upgrading.
+
+A Kyverno policy for `provider-kairos` images might look like the following:
+
+```yaml
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-image
+spec:
+  validationFailureAction: Enforce
+  background: false
+  webhookTimeoutSeconds: 30
+  failurePolicy: Fail
+  rules:
+    - name: check-image
+      match:
+        any:
+        - resources:
+            kinds:
+              - Pod
+      verifyImages:
+      - imageReferences:
+        - "quay.io/kairos/kairos-*"
+        attestors:
+        - entries:
+          # See: https://kyverno.io/docs/writing-policies/verify-images/#keyless-signing-and-verification
+          - keyless:
+              subject: "https://github.com/kairos-io/provider-kairos/.github/workflows/release.yaml@refs/tags/*"
```

Comment:
this is kind of icky.

Hear me out, this is really good until cosign or the CI breaks and we need to sign the already published artifacts in a different way, maybe by a different workflows like we used to do in that other project. Then this doesnt match anymore so you either drop a release completely as it wont be signed, resign it and break this gating for users or run a new release and left the unsigned release there.

does this subject allows for regex? or for value.in so we can provide a list?

Reply:
the subject is a regex already indeed: `https://github.com/kairos-io/provider-kairos/.github/workflows/release.yaml@refs/tags/*` I think it's fine as a baseline example and nothing more, for instance, policy can be applied also per-releases prior applying a plan, and so on so forth